vnext/ReactUWP/Views/ViewControl.cpp

```diff
@@ -29,70 +22,152 @@ ViewControl::ViewControl()
-        case ::react::uwp::AccessibilityRoles::Unknown:
-        default:
-          return winrt::Windows::UI::Xaml::Automation::Peers::AutomationControlType::Group;
-      }
+winrt::IInspectable DynamicAutomationPeer::GetPatternCore(winrt::PatternInterface const& patternInterface) const
+{
+  if (auto viewControl = Owner().try_as<ViewControl>())
+  {
+    auto accessibilityRole = viewControl->AccessibilityRole();
+
+    if (patternInterface == winrt::PatternInterface::Invoke &&
+      (accessibilityRole == AccessibilityRoles::Button || accessibilityRole == AccessibilityRoles::ImageButton))
+    {
+      return *this;
     }
+    else if (patternInterface == winrt::PatternInterface::Selection || patternInterface == winrt::PatternInterface::SelectionItem)
+    {
+      return *this;
+    }
+  }
+
+  return nullptr;
+}
+
+bool DynamicAutomationPeer::IsEnabledCore() const
+{
+  bool disabled = false;
+
+  if (auto viewControl = Owner().try_as<ViewControl>())
+  {
+    disabled = viewControl->AccessibilityState(AccessibilityStates::Disabled);
+  }
+
+  return !disabled && Super::IsEnabledCore();
+}
+
+// IInvokeProvider
 
-    Windows::Foundation::IInspectable DynamicAutomationPeer::GetPatternCore(Windows::UI::Xaml::Automation::Peers::PatternInterface const& patternInterface) const
+void DynamicAutomationPeer::Invoke() const
+{
+  if (auto viewControl = Owner().try_as<ViewControl>())
+  {
+    if (auto invokeHandler = viewControl->AccessibilityInvoke())
     {
-      auto viewControl = Owner().as<::react::uwp::ViewControl>();
+      invokeHandler();
+    }
+  }
+}
 
-      if (patternInterface == Windows::UI::Xaml::Automation::Peers::PatternInterface::Invoke &&
-          (viewControl->AccessibilityRole() == ::react::uwp::AccessibilityRoles::Button || 
-           viewControl->AccessibilityRole() == ::react::uwp::AccessibilityRoles::ImageButton))
-      {
-        return *this;
-      }
+// ISelectionProvider
+
+winrt::com_array<winrt::IRawElementProviderSimple> DynamicAutomationPeer::GetSelection() const
+{
+  return {};
+}
 
-      return nullptr;
+// ISelectionItemProvider
+
+bool DynamicAutomationPeer::IsSelected() const
+{
+  auto viewControl = Owner().try_as<ViewControl>();
+  return (nullptr != viewControl && viewControl->AccessibilityState(AccessibilityStates::Selected));
+}
+
+winrt::IRawElementProviderSimple DynamicAutomationPeer::SelectionContainer() const
+{
+  if (auto viewControl = GetParentViewControl())
+  {
+    if (auto peer = winrt::FrameworkElementAutomationPeer::CreatePeerForElement(viewControl.as<winrt::UIElement>()))
+    {
+      return ProviderFromPeer(peer);
     }
+  }
+
+  return nullptr;
+}
+
+void DynamicAutomationPeer::AddToSelection() const
+{
+  
+}
 
+void DynamicAutomationPeer::RemoveFromSelection() const
+{
+
+}
+
+void DynamicAutomationPeer::Select() const
+{
+
+}
 
-    void DynamicAutomationPeer::Invoke() const
+winrt::com_ptr<ViewControl> DynamicAutomationPeer::GetParentViewControl() const
```

Comment:
>winrt::com_ptr<ViewControl> [](start = 0, length = 27)

can we use winrt::ViewControl? I didn't see any reason to wrap as com_ptr here.

Reply:
Because it's not an exposed type, just an impl type (winrt::com_impl<ViewControl>), I have to wrap it in a com_ptr in order to return it.